### PR TITLE
Replace app name check with regex

### PR DESCRIPTION
**Pull request by kimlisa**
resolves https://github.com/gravitational/teleport/issues/4953

#### Description
- replace URL object app name check with regex (same regex as the backend uses):
https://github.com/kubernetes/apimachinery/blob/release-1.20/pkg/util/validation/validation.go#L219

### CODE REVIEW

File: packages/teleport/src/Apps/AddApp/Automatically/Automatically.tsx [hunk 1]
```diff
@@ -180,20 +189,18 @@ const requiredAppName = value => () => {
...
-    if (tmp.hostname.split('.').length > 1) {
-      throw new Error();
-    }
-  } catch {
+  const match = value.match(REGEX_DNS1035LABEL);
+  if (!match) {
     return {
       valid: false,
-      message: 'Invalid',
+      message: 'Invalid name',
```
Comment (alex-kovoy) [inline packages/teleport/src/Apps/AddApp/Automatically/Automatically.tsx:203]: lets change it to `Should be a valid DNS sub-domain name`

Comment (alex-kovoy) [reply packages/teleport/src/Apps/AddApp/Automatically/Automatically.tsx:203]: could you also ensure that the text inside app name field is also a lower case? 

I think the css rule would do.

```
text-transform: lowercase;
```

**Review by alex-kovoy — APPROVED** [review]
with comments

File: packages/teleport/src/Apps/AddApp/Automatically/Automatically.tsx (hunk 1 above)
Comment (kimlisa) [reply packages/teleport/src/Apps/AddApp/Automatically/Automatically.tsx]: revised as discussed in slack

**Pull request merged**